src/jemalloc.c

```diff
@@ -3275,14 +3278,11 @@ do_rallocx(void *ptr, size_t size, int flags, bool is_realloc) {
 	UTRACE(ptr, size, p);
 	check_entry_exit_locking(tsd_tsdn(tsd));
 
-	if (config_fill && malloc_slow && !zero && usize > old_usize) {
+	if (config_fill && unlikely(opt_junk_alloc) && usize > old_usize
+	    && !zero) {
```

Comment:
Why there is no `malloc_slow`?

Reply:
opt_junk_alloc implies malloc_slow, but we'd still need to check it inside the branch (previously, checking it was helpful, since we folded both branches behind malloc_slow).